src/components/Address/Address.js

```diff
@@ -0,0 +1,154 @@
+import React, { useState, useCallback } from 'react'
+import PropTypes from 'prop-types'
+import ENS from 'ethereum-ens'
+import _ from 'lodash'
+import { getEthAddressType, isAddress, ETH_ADDRESS_TYPE } from '../../utils/address.js'
+import Loader from '../../components/Loader.js'
+import { SingleNameBlockies } from '../../components/SingleName/SingleNameBlockies.js'
+import warningImage from '../../assets/warning.svg'
+import searchImage from '../../assets/search.svg'
+
+import './style.css'
+
+const ENS_NOT_FOUND = 'ENS name not found'
+
+function Address(props) {
+  const [resolvedAddress, setResolvedAddress] = useState(null)
+  const [inputValue, setInputValue] = useState('')
+  const [isResolvingInProgress, setIsResolvingInProgress] = useState(false)
+  const [error, setError] = useState(null)
+
+  const ens = new ENS(props.provider)
```

Comment:
It's probably not a good idea to reinstantiate this every time we render. We should probably only do this once

Reply:
You're right. Fixed.